modules/cockroach/src/main/java/org/testcontainers/containers/CockroachContainer.java

```diff
@@ -0,0 +1,95 @@
+package org.testcontainers.containers;
+
+import org.testcontainers.containers.wait.HttpWaitStrategy;
+
+import java.time.Duration;
+
+public class CockroachContainer<SELF extends CockroachContainer<SELF>> extends JdbcDatabaseContainer<SELF> {
+    public static final String NAME = "cockroach";
+    public static final String IMAGE = "cockroachdb/cockroach";
+    public static final String IMAGE_TAG = "v1.1.2";
+    public static final String JDBC_DRIVER_CLASS_NAME = "org.postgresql.Driver";
+    public static final String JDBC_URL_PREFIX = "jdbc:postgresql";
+    public static final String TEST_QUERY_STRING = "SELECT 1";
+    public static final int REST_API_PORT = 8080;
+    public static final int DB_PORT = 26257;
+
+    private String databaseName = "test";
+    private String username = "root";
+    private String password = "";
+
+    public CockroachContainer() {
+        this(IMAGE + ":" + IMAGE_TAG);
+    }
+
+    public CockroachContainer(final String dockerImageName) {
+        super(dockerImageName);
+    }
+
+    @Override
+    protected Integer getLivenessCheckPort() {
+        return getMappedPort(REST_API_PORT);
+    }
+
+    @Override
+    protected void configure() {
```

Comment:
I would move most of them to the constructor, so that the end user can override some parameters without extending the class